### PR TITLE
Username edit visual fixes

### PR DESCRIPTION
**Pull request by JPonte**
#### APK
[Download build #8051](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8051/artifact/build/artifact/wire-dev-PR379-8051.apk)
[Download build #8054](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8054/artifact/build/artifact/wire-dev-PR379-8054.apk)
[Download build #8059](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8059/artifact/build/artifact/wire-dev-PR379-8059.apk)

### CODE REVIEW

**Comment by JPonte** [comment]
build again

File: wire-core/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -432,7 +432,9 @@
     <dimen name="prefs__add_email_pw__margin_textinput">16dp</dimen>
     <dimen name="prefs__username_change__margins_between">24dp</dimen>
     <dimen name="prefs__username_change__edit_text_horizontal_padding">28dp</dimen>
+    <dimen name="prefs__username_change__at_top_padding">16dp</dimen>
```
Comment (shumengye) [inline wire-core/src/main/res/values/dimens.xml]: Replace 16dp and 24dp with standard wire padding values in `dimens.xml`

File: app/src/main/java/com/waz/zclient/pages/main/profile/preferences/dialogs/ChangeUsernamePreferenceDialogFragment.java [hunk 2]
```diff
@@ -157,6 +182,21 @@ public static ChangeUsernamePreferenceDialogFragment newInstance(String currentU
     }
 
     @Override
+    public void onStart() {
```
Comment (shumengye) [inline app/src/main/java/com/waz/zclient/pages/main/profile/preferences/dialogs/ChangeUsernamePreferenceDialogFragment.java]: Put `onStart()` after `onCreate()` lifecycle events

File: app/src/main/java/com/waz/zclient/pages/main/profile/preferences/dialogs/ChangeUsernamePreferenceDialogFragment.java [hunk 3]
```diff
@@ -49,25 +54,41 @@
     private View backButton;
     private String inputUsername = "";
     private boolean editingEnabled = true;
+    private Self self = null;
+
+    private ModelObserver<Self> selfModelObserver = new ModelObserver<Self>() {
```
Comment (shumengye) [inline app/src/main/java/com/waz/zclient/pages/main/profile/preferences/dialogs/ChangeUsernamePreferenceDialogFragment.java]: make sure to stop observing the model at some point

**Comment by shumengye** [comment]
@JPonte 
Generally be careful not to add `OnClickListener` in `onCreate()`, better to do add and remove in `onPause()` and `onResume()`. Else be careful to check that referenced views, stores and controllers still exist.

File: app/src/main/java/com/waz/zclient/pages/main/profile/preferences/dialogs/ChangeUsernamePreferenceDialogFragment.java [hunk 4]
```diff
@@ -157,6 +182,21 @@ public static ChangeUsernamePreferenceDialogFragment newInstance(String currentU
     }
 
     @Override
+    public void onStart() {
+        super.onStart();
+        getView().post(new Runnable() {
```
Comment (shumengye) [inline app/src/main/java/com/waz/zclient/pages/main/profile/preferences/dialogs/ChangeUsernamePreferenceDialogFragment.java]: Check how keyboard is triggered when editing e.g. email in the settings

File: wire-core/src/main/res/values/dimens.xml (hunk 1 above)
Comment (JPonte) [inline wire-core/src/main/res/values/dimens.xml]: This one is a very specific padding for one element, it shouldn't inherit the standard wire padding.

File: app/src/main/java/com/waz/zclient/pages/main/profile/preferences/dialogs/ChangeUsernamePreferenceDialogFragment.java (hunk 4 above)
Comment (JPonte) [inline app/src/main/java/com/waz/zclient/pages/main/profile/preferences/dialogs/ChangeUsernamePreferenceDialogFragment.java]: Forget it, found out what the issue was. Won't need to post it after all.

**Pull request merged**